pkg/controller/node/nodecontroller.go

```diff
@@ -480,9 +480,26 @@ func (nc *NodeController) monitorNodeStatus() error {
 						continue
 					}
 					if remaining {
-						// queue eviction of the pods on the node
+						// Immediately evict pods (skip rate-limited evictor)
 						glog.V(2).Infof("Deleting node %s is delayed while pods are evicted", node.Name)
-						nc.evictPods(node.Name)
+						go func(nodeName string) {
+							nc.evictorLock.Lock()
+							defer nc.evictorLock.Unlock()
+							remaining, err := nc.deletePods(nodeName)
+							if err != nil {
+								glog.Errorf("Unable to evict pods from node %s: %v", nodeName, err)
+								nc.podEvictor.Add(nodeName)
+								return
+							}
+							if !remaining {
+								return
+							}
+							// Immediately terminate pods.
```

Comment:
Yup - I don't think that rate limiting makes any sense here. This is a branch responsible for evicting Pods from Nodes that are gone from the cloud provider's perspective. If for whatever reason we would be able to contact this Node it would mean that there's some serious bug in the control plane of the given provider.